compiler/crates/relay-lsp/src/rename/mod.rs

```diff
@@ -0,0 +1,399 @@
+
+                                let location = LspLocation::new(old_file_uri.clone(), name_range);
+
+                                rename_operation(new_operation_name, location)
+                            }
+                            ExecutableDefinition::Operation(
+                                graphql_syntax::OperationDefinition { name: None, .. },
+                            ) => HashMap::new(),
+                        };
+
+                        merge_text_changes(&mut rename_changes, changes);
+                    }
+                }
+                _ => (),
+            };
+        }
+    }
+
+    Ok(Some(WorkspaceEdit {
+        changes: Some(rename_changes),
+        ..Default::default()
+    }))
+}
+
+fn rename_relay_resolver_field(
+    field_name: StringKey,
+    type_name: StringKey,
+    new_field_name: &str,
+    program: &Program,
+    schema: &Arc<SDLSchema>,
+    root_dir: &PathBuf,
+) -> LSPRuntimeResult<HashMap<Url, Vec<TextEdit>>> {
+    let mut changes = HashMap::<Url, Vec<TextEdit>>::new();
+
+    for location in
+        find_field_locations(program, schema, field_name, type_name).unwrap_or_else(|| vec![])
+    {
+        let lsp_location = transform_relay_location_to_lsp_location(root_dir, location)?;
+
+        merge_text_edit(&mut changes, lsp_location, new_field_name);
+    }
+
+    Ok(changes)
+}
+
+fn rename_operation(
+    new_operation_name: String,
+    location: LspLocation,
+) -> HashMap<Url, Vec<TextEdit>> {
+    HashMap::from([(
+        location.uri,
+        vec![TextEdit {
+            new_text: new_operation_name,
+            range: location.range,
+        }],
+    )])
+}
+
+fn rename_fragment(
+    fragment_name: StringKey,
+    new_fragment_name: String,
+    program: &Program,
+    root_dir: &PathBuf,
+) -> LSPRuntimeResult<HashMap<Url, Vec<TextEdit>>> {
+    let mut changes = HashMap::<Url, Vec<TextEdit>>::new();
+
+    for location in FragmentFinder::get_fragment_usages(program, fragment_name) {
+        let lsp_location = transform_relay_location_to_lsp_location(root_dir, location)?;
+        merge_text_edit(&mut changes, lsp_location, &new_fragment_name);
+    }
+
+    Ok(changes)
+}
+
+fn extract_parent_type(docblock: DocblockIr) -> StringKey {
+    match docblock {
+        DocblockIr::RelayResolver(resolver_ir) => match resolver_ir.on {
+            On::Type(on_type) => on_type.value.item,
+            On::Interface(on_interface) => on_interface.value.item,
+        },
+        DocblockIr::TerseRelayResolver(resolver_ir) => resolver_ir.type_.item,
+        DocblockIr::StrongObjectResolver(strong_object) => strong_object.type_name.value,
+        DocblockIr::WeakObjectType(weak_type_ir) => weak_type_ir.type_name.value,
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct FragmentFinder {
+    fragment_locations: Vec<IRLocation>,
+    fragment_name: StringKey,
+}
+
+impl FragmentFinder {
+    pub fn get_fragment_usages(program: &Program, name: StringKey) -> Vec<IRLocation> {
```

Comment:
Would it be possible to reuse https://github.com/facebook/relay/blob/main/compiler/crates/relay-lsp/src/references/mod.rs#L119?